### ISSUE TITLE
Misunderstanding of Receiver Assignment Propagation in `MigrateLegacy`

### ISSUE THREAD

**Issue by naveensrinivasan** [defenseunicorns/zarf#2348]
### Issue Body:
I've identified a potential misunderstanding regarding how receiver assignments within a method affect the caller's instance, particularly in the context of the `MigrateLegacy`method in `package.go`. This misunderstanding revolves around the expectation that reassigning the method receiver (e.g., `pp = pp.AddSBOMs()`) would propagate back to the caller, which is not aligned with Go's behavior regarding method receivers.
https://github.com/defenseunicorns/zarf/blob/4d8b8339c010c9c82385cd6748ded9c4048a2b57/src/pkg/layout/package.go#L79-L89

#### File: 
`package.go`

#### Function:
`MigrateLegacy`

#### Detailed Explanation:
In Go, when a method receiver is reassigned within the method, this reassignment does not affect the caller's instance. This behavior can be illustrated with a simple example:

```go
package main

import "fmt"

type MyStruct struct {
    Value int
}

// Method with a pointer receiver
func (m *MyStruct) PointerReceiverMethod() {
    // This modification affects the original.
    m.Value = 10

    // This reassignment does NOT affect the original pointer.
    m = &MyStruct{Value: 20}
}

// Method with a value receiver
func (m MyStruct) ValueReceiverMethod() {
    // This modification only affects the copy within this method.
    m.Value = 30
}

func main() {
    original := &MyStruct{Value: 0}
    original.PointerReceiverMethod()
    fmt.Println(original.Value) // Outputs: 10, because the field modification is visible.

    original.ValueReceiverMethod()
    fmt.Println(original.Value) // Still outputs: 10, modifications in a value receiver method do not propagate.
}
```

In this example, modifications to the receiver's fields within `PointerReceiverMethod` are visible to the caller because it operates directly on the original instance. However, reassigning the receiver to point to a new instance (`m = &MyStruct{Value: 20}`) within the method does not change the caller's instance. This reassignment is local to the method's scope.

#### Issue in `MigrateLegacy`:
The `MigrateLegacy` method contains lines like `pp = pp.AddSBOMs()`, suggesting an expectation that such reassignments would propagate back to the caller. This is a misunderstanding of Go's semantics, as such reassignments only affect the method's local scope.

#### Suggested Action:
It's crucial to clarify the intent behind these receiver reassignments in the `MigrateLegacy` method. If the intention is to modify the caller's instance, the current approach may not achieve the desired effect. A review and possible revision of the method's implementation could ensure it aligns with Go's behavior and the intended outcomes.

This clarification will help prevent confusion and ensure that the method's behavior is correctly understood and utilized by developers working with or extending the codebase.

**Comment by Noxsios** [comment]
@naveensrinivasan This behavior is really a quick and dirty implementation I wrote for modular population of a struct at runtime. I chose reassignment over `pp.AddSBOMs()` to be a bit more clear that the operation was mutating the internals of the struct itself. Whatever is the true go-native / semantic pattern to implement here I am all for it!

**Comment by lucasrod16** [comment]
@naveensrinivasan and I discussed this issue synchronously. We verified that the behavior of the client/calling code mentioned in this issue works as expected. We acknowledge that `PackagePaths` receiver methods such as `AddImages` and `AddSBOMs` modify the state of the `PackagePaths` object _and then_ returns the modified object, and this is not strictly necessary or idiomatic Go. With that being said, we are not prioritizing updating our codebase to be idiomatic at the moment. Because this is low priority for the Zarf team, I am closing this issue for now.
